rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/dataspace/FileSystem.java

```diff
@@ -104,26 +98,44 @@ public FileObject createFile(String pathname) throws FileSystemException {
         return fo;
     }
 
-    public static ListFile list(FileObject fo) throws FileSystemException {
+    public static ListFile list(FileObject fo, List<String> includes, List<String> excludes) throws FileSystemException {
         fo.refresh();
         ListFile list = new ListFile();
         List<String> dirList = Lists.newArrayList();
         List<String> fileList = Lists.newArrayList();
-        for (FileObject child : fo.getChildren()) {
+        List<String> fullList = Lists.newArrayList();
+        List<FileObject> foundFileObjects = new LinkedList<>();
+        if (isNullOrEmpty(includes) && isNullOrEmpty(excludes)) {
+            fo.findFiles(Selectors.SELECT_CHILDREN, false, foundFileObjects);
+        } else {
+            FileSelector selector = new org.objectweb.proactive.extensions.dataspaces.vfs.selector.FileSelector(includes, excludes);
+            fo.findFiles(selector, false, foundFileObjects);
+        }
+
+        for (FileObject child : foundFileObjects) {
             FileType type = child.getType();
+            FileName childName = child.getName();
             switch (type) {
                 case FOLDER:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1151)